tests/testsuite/update.rs

```diff
@@ -2357,31 +2357,31 @@ fn update_breaking_spec_version_transitive() {
                 authors  =  []
 
                 [dependencies]
-                dep  =  "1.1"
+                dep  =  "2.0"
             "#,
         )
         .file("bar/src/lib.rs", "")
         .build();
 
     p.cargo("generate-lockfile").run();
 
-    Package::new("dep", "1.1.1").publish();
-    Package::new("dep", "2.0.0").publish();
+    Package::new("dep", "2.0.1").publish();
+    Package::new("dep", "3.0.0").publish();
 
     // Will upgrade the direct dependency
     p.cargo("update -Zunstable-options --breaking dep@1.0")
         .masquerade_as_nightly_cargo(&["update-breaking"])
         .with_stderr_data(str![[r#"
 [UPDATING] `[..]` index
-[UPGRADING] dep ^1.0 -> ^2.0
+[UPGRADING] dep ^1.0 -> ^3.0
```

Comment:
There is discrepancy between the normal update and breaking update. 

* With and without `v` prefix
* Show SemVer requirement operator or not.

Not a blocker for this PR, but we might want  track this somewhere.

Reply:
This was discussed here: https://github.com/rust-lang/cargo/pull/13979#discussion_r1628470396

I'll add it to the [task list](https://github.com/rust-lang/cargo/issues/12425#issuecomment-2186198258).